// tv-recommender-mcp-server/dist/tools/index.js
"use strict";
/**
 * MCP 工具导出文件
 * 集中导出所有MCP工具函数
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getShowVideos = exports.getTrendingShows = exports.getPopularShows = exports.getShowReviews = exports.getActorDetailsAndCredits = exports.findPersonId = exports.getRecommendationsByActor = exports.findShowsByPersonName = exports.discoverShows = exports.getWatchProviders = exports.getShowDetails = exports.getSimilarShows = exports.getRecommendationsByGenre = void 0;
// 导出推荐相关工具
var recommendations_1 = require("./recommendations");
Object.defineProperty(exports, "getRecommendationsByGenre", { enumerable: true, get: function () { return recommendations_1.getRecommendationsByGenre; } });
Object.defineProperty(exports, "getSimilarShows", { enumerable: true, get: function () { return recommendations_1.getSimilarShows; } });
// 导出剧集详情工具
var details_1 = require("./details");
Object.defineProperty(exports, "getShowDetails", { enumerable: true, get: function () { return details_1.getShowDetails; } });
// 导出观看渠道查询工具
var watchProviders_1 = require("./watchProviders");
Object.defineProperty(exports, "getWatchProviders", { enumerable: true, get: function () { return watchProviders_1.getWatchProviders; } });
// 导出高级发现工具
var discoverShowsTool_1 = require("./discoverShowsTool");
Object.defineProperty(exports, "discoverShows", { enumerable: true, get: function () { return discoverShowsTool_1.discoverShows; } });
Object.defineProperty(exports, "findShowsByPersonName", { enumerable: true, get: function () { return discoverShowsTool_1.findShowsByPersonName; } });
Object.defineProperty(exports, "getRecommendationsByActor", { enumerable: true, get: function () { return discoverShowsTool_1.getRecommendationsByActor; } });
Object.defineProperty(exports, "findPersonId", { enumerable: true, get: function () { return discoverShowsTool_1.findPersonId; } });
// 导出演员信息工具
var actorInfoTool_1 = require("./actorInfoTool");
Object.defineProperty(exports, "getActorDetailsAndCredits", { enumerable: true, get: function () { return actorInfoTool_1.getActorDetailsAndCredits; } });
// 导出评论工具
var reviewsTool_1 = require("./reviewsTool");
Object.defineProperty(exports, "getShowReviews", { enumerable: true, get: function () { return reviewsTool_1.getShowReviews; } });
// 导出热门与趋势剧集工具
var popularTrendingTool_1 = require("./popularTrendingTool");
Object.defineProperty(exports, "getPopularShows", { enumerable: true, get: function () { return popularTrendingTool_1.getPopularShows; } });
Object.defineProperty(exports, "getTrendingShows", { enumerable: true, get: function () { return popularTrendingTool_1.getTrendingShows; } });
// 导出视频工具
var videosTool_1 = require("./videosTool");
Object.defineProperty(exports, "getShowVideos", { enumerable: true, get: function () { return videosTool_1.getShowVideos; } });
